include/seqan3/io/sequence/sequence_file_format_fasta.hpp

```diff
@@ -6,61 +43,103 @@
+                 sequence_of_sequence_concept<std::decay_t<quals_type>>
     void read(seqs_type && seqs,
-              ids_type && ids,
+              metas_type && metas,
               quals_type && quals,
               stream_type & stream,
-              options_type const & options);
-};
+              options_type const & options,
+              size_t max_records = 0);
 
-static_assert(align_file_out_format_concept<align_file_out_format_sam>,
-              "align_file_out_format_sam does not satisfy align_file_out_format_concept");
+    template <typename sequence_type,
+              typename meta_type,
+              typename qual_type,
+              typename stream_type,
+              typename options_type>
+        requires sequence_concept<std::decay_t<sequence_type>> &&
+                 sequence_concept<std::decay_t<meta_type>> &&
+                 sequence_concept<std::decay_t<qual_type>>
+    void write(sequence_type && seq,
+              meta_type && meta,
+              qual_type && qual,
+              stream_type & stream,
+              options_type const & options);
 
-/** implementations **/
+    template <typename seqs_type,
+              typename metas_type,
+              typename quals_type,
+              typename stream_type,
+              typename options_type>
+              requires sequence_of_sequence_concept<std::decay_t<seqs_type>> &&
+                       sequence_of_sequence_concept<std::decay_t<metas_type>> &&
+                       sequence_of_sequence_concept<std::decay_t<quals_type>>
+    void write(seqs_type && seqs,
+               metas_type && metas,
+               quals_type && quals,
+               stream_type & stream,
+               options_type const & options,
+               size_t max_records = 0);
+};
 
+template <typename sequence_type,
+          typename meta_type,
+          typename qual_type,
+          typename stream_type,
+          typename options_type>
+     requires sequence_concept<std::decay_t<sequence_type>> &&
+              sequence_concept<std::decay_t<meta_type>> &&
+              sequence_concept<std::decay_t<qual_type>>
+void sequence_file_format_fasta::read(sequence_type && seq,
+                                      meta_type && meta,
+                                      qual_type && qual,
+                                      stream_type & stream,
+                                      options_type const & options)
+{
+    // TODO:: tokenization
+}
 
-} // namespace seqan3
\ No newline at end of file
+template <typename seqs_type,
+          typename metas_type,
+          typename quals_type,
+          typename stream_type,
+          typename options_type>
+    requires sequence_of_sequence_concept<std::decay_t<seqs_type>> &&
+             sequence_of_sequence_concept<std::decay_t<metas_type>> &&
+             sequence_of_sequence_concept<std::decay_t<quals_type>>
+void sequence_file_format_fasta::read(seqs_type && seqs,
```

Comment:
inline and put sequence name on separate line. Please do this for all interfaces